pkg/scheduler/algorithm/priorities/resource_bin_packing.go

```diff
@@ -0,0 +1,240 @@
+		{
+			for i := range pod.Spec.Containers {
+				container := &pod.Spec.Containers[i]
+				_, memory := priorityutil.GetNonzeroRequests(&container.Resources.Requests)
+				requested.Memory += memory
+			}
+			for i := range pod.Spec.InitContainers {
+				container := &pod.Spec.InitContainers[i]
+				_, memory := priorityutil.GetNonzeroRequests(&container.Resources.Requests)
+				if requestedInit.Memory < memory {
+					requestedInit.Memory = memory
+				}
+			}
+			if requestedInit.Memory > requested.Memory {
+				requested.Memory = requestedInit.Memory
+			}
+			//println("Memory used: %d, requested: %d, available: %d",
+			//	nodeInfo.NonZeroRequest().Memory, requested.Memory, allocatable.Memory)
+			return float64((nodeInfo.NonZeroRequest().Memory+requested.Memory)*schedulerapi.MaxPriority) / float64(allocatable.Memory)
+		}
+
+	case v1.ResourceEphemeralStorage:
+		{
+			for i := range pod.Spec.Containers {
+				container := &pod.Spec.Containers[i]
+				if quantity, ok := container.Resources.Requests[v1.ResourceEphemeralStorage]; ok {
+					requested.EphemeralStorage += quantity.Value()
+				}
+
+			}
+			for i := range pod.Spec.InitContainers {
+				container := &pod.Spec.InitContainers[i]
+				if quantity, ok := container.Resources.Requests[v1.ResourceEphemeralStorage]; ok {
+					if requestedInit.EphemeralStorage < quantity.Value() {
+						requestedInit.EphemeralStorage = quantity.Value()
+					}
+				}
+			}
+			if requestedInit.EphemeralStorage > requested.EphemeralStorage {
+				requested.EphemeralStorage = requestedInit.EphemeralStorage
+			}
+			return float64((used.EphemeralStorage+requested.EphemeralStorage)*schedulerapi.MaxPriority) / float64(allocatable.EphemeralStorage)
+
+		}
+	default:
+		if v1helper.IsScalarResourceName(resource) {
+			if podRequestsResource(*pod, resource) {
+				for i := range pod.Spec.Containers {
+					container := &pod.Spec.Containers[i]
+					if quantity, ok := container.Resources.Requests[resource]; ok {
+						requested.AddScalar(resource, quantity.Value())
+					}
+				}
+				for i := range pod.Spec.InitContainers {
+					container := &pod.Spec.InitContainers[i]
+					if quantity, ok := container.Resources.Requests[resource]; ok {
+						if requestedInit.ScalarResources[resource] < quantity.Value() {
+							requested.SetScalar(resource, quantity.Value())
+						}
+					}
+
+				}
+				if requestedInit.ScalarResources[resource] > requested.ScalarResources[resource] {
+					requested.ScalarResources[resource] = requestedInit.ScalarResources[resource]
+				}
+
+				return float64((used.ScalarResources[resource]+requested.ScalarResources[resource])*schedulerapi.MaxPriority) / float64(allocatable.ScalarResources[resource])
+
+			}
+		}
+
+	}
+	return 0
+}
+
+// podRequestsResource checks if a given pod requests the scare resource. if false the priority is set to 0
+func podRequestsResource(pod v1.Pod, resource v1.ResourceName) bool {
+	if strings.Contains(string(resource), "cpu") {
```

Comment:
If we want to check `cpu`/`memory`, please use equal instead of `Contains` which may introduce unexpected result.

Reply:
also i dont understand this check. Shouldnt the function be checking , if `resource` is present as one of the requested resource inside Pod ?

Reply:
I updated the code to not check for memory or CPU just extended resources. If we program our function to bin pack over extended resources it is not necessary that most request would contain the extended resource. So we can avoid calculating the score for such pods. It does not make sense to check if we bin pack over memory or CPU as all pods have some default request for CPU or Memory